woocommerce/woocommerce
# Abstract PR scrub action so it can be used in two places

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This PR abstracts the code to send scrub notifications out of `test-assistant-pr-scrub.yml` and into a new composite action. This is done because, while `pull-request-post-merge-processing.yml` adds the label `needs: analysis`, labels added by GitHub actions do not trigger subsequent GitHub actions. Rather than add the need to manage yet another GitHub token, I've abstracted the functionality and called it from both workflows.





Closes # .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Fork this repository and merge the changes in this PR to trunk within your fork.
2. Enable issues in your fork.
3. Enable actions in your fork.
4. The following steps for testing Slack were borrowed from #38290, with slight modification:
5. I think the nicest way to test this without interfering with a real Slack instance is to [create a new slack workspace](https://slack.com/get-started#/createnew).
6. Now log in to the workspace and [create an app](https://api.slack.com/authentication/basics#creating).
7. In the app settings you need to [give it permissions](https://api.slack.com/authentication/basics#scopes) to send messages to generate a token. chat:write should be sufficient.
8. Once you give the scopes, save the [auth token](https://api.slack.com/authentication/basics#getting-your-authentication-token) generated in your password manager of choice.
9. Configure 2 secrets in your fork: `TEST_ASSISTANCE_BOT_TOKEN` should contain the auth token from 8.`WOO_PR_TESTING_ANALYSIS_SLACK_CHANNEL` should be `#general` because every slack instance has this channel by default. `WOO_PR_TESTING_ANALYSIS_TEAM_ID` isn't important for testing.
10. Create a PR in your fork and add the `needs: analysis` label. Verify that the action is run and a message about the PR is sent to your channel that indicates that the PR is **open**.
11. Merge the PR you created in the previous step. No message should be sent because it was merged already containing the label, and a message was already sent about analysis.
12. Create a new PR in your fork within the WooCommerce plugin, this time do not add the label. The action should run and your channel should receive a message indicating that the PR is **merged**.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
